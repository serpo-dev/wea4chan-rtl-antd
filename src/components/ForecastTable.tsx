import { Timeline } from "antd";
import { FC } from "react";

interface ForecastTableProps {
    hourly: {
        temperature_2m: Array<number>;
        time: Array<string>;
    };
    curHour: number;
}

function getItemStyle({ hour, curHour }: { hour: number; curHour: number }) {
    console.log(hour, curHour);
    if (hour > curHour) return { color: "blue" };
    if (hour < curHour) return { color: "gray" };
    return { color: "green" };
}

export const ForecastTable: FC<ForecastTableProps> = ({
    hourly,
    curHour,
}: ForecastTableProps) => {
    const items = hourly?.time?.map((t, i) => ({
        label: t.slice(11),
        children: (
            <div style={{ fontWeight: 500 }}>
                {hourly?.temperature_2m[i] + " Â°C"}
            </div>
        ),
        ...getItemStyle({ hour: new Date(t).getHours(), curHour }),
    }));

    return (
        <div
            style={{
                display: "flex",
                flexDirection: "row",
                marginTop: "60px",
                width: "500px",
            }}
        >
            <Timeline
                style={{ flex: 1 }}
                mode="right"
                items={items.slice(0, 12)}
            />
            <Timeline
                style={{ flex: 1 }}
                mode="left"
                items={items.slice(12, 24)}
            />
        </div>
    );
};
