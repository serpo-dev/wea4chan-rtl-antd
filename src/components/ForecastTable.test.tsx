import { render } from "@testing-library/react";
import { ForecastTable } from "./ForecastTable";

const mockCurHour = 12;
const mockHourly = {
    time: [
        "2023-06-20T00:00",
        "2023-06-20T01:00",
        "2023-06-20T02:00",
        "2023-06-20T03:00",
        "2023-06-20T04:00",
        "2023-06-20T05:00",
        "2023-06-20T06:00",
        "2023-06-20T07:00",
        "2023-06-20T08:00",
        "2023-06-20T09:00",
        "2023-06-20T10:00",
        "2023-06-20T11:00",
        "2023-06-20T12:00",
    ],
    temperature_2m: [
        19, 19.3, 19.6, 19.9, 20, 19.9, 19.7, 19.3, 19.2, 19.2, 18.7, 18.5,
        18.3, 18.1, 18.2, 18.1,
    ],
};

describe("should render ForecastTable properly", () => {
    it("should match the snapshot", () => {
        const { asFragment } = render(
            <ForecastTable
                hourly={mockHourly}
                curHour={mockCurHour}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
