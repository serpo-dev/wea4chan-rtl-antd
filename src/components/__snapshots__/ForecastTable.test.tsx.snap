// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ForecastTable properly should match the snapshot 1`] = `
<DocumentFragment>
  <div
    style="display: flex; flex-direction: row; margin-top: 60px; width: 500px;"
  >
    <ul
      class="ant-timeline ant-timeline-label css-dev-only-do-not-override-12jzuas"
      style="flex: 1;"
    >
      <li
        class="ant-timeline-item ant-timeline-item-right"
      >
        <div
          class="ant-timeline-item-label"
        >
          00:00
        </div>
        <div
          class="ant-timeline-item-tail"
        />
        <div
          class="ant-timeline-item-head ant-timeline-item-head-gray"
        />
        <div
          class="ant-timeline-item-content"
        >
          <div
            style="font-weight: 500;"
          >
            19 °C
          </div>
        </div>
      </li>
      <li
        class="ant-timeline-item ant-timeline-item-right"
      >
        <div
          class="ant-timeline-item-label"
        >
          01:00
        </div>
        <div
          class="ant-timeline-item-tail"
        />
        <div
          class="ant-timeline-item-head ant-timeline-item-head-gray"
        />
        <div
          class="ant-timeline-item-content"
        >
          <div
            style="font-weight: 500;"
          >
            19.3 °C
          </div>
        </div>
      </li>
      <li
        class="ant-timeline-item ant-timeline-item-right"
      >
        <div
          class="ant-timeline-item-label"
        >
          02:00
        </div>
        <div
          class="ant-timeline-item-tail"
        />
        <div
          class="ant-timeline-item-head ant-timeline-item-head-gray"
        />
        <div
          class="ant-timeline-item-content"
        >
          <div
            style="font-weight: 500;"
          >
            19.6 °C
          </div>
        </div>
      </li>
      <li
        class="ant-timeline-item ant-timeline-item-right"
      >
        <div
          class="ant-timeline-item-label"
        >
          03:00
        </div>
        <div
          class="ant-timeline-item-tail"
        />
        <div
          class="ant-timeline-item-head ant-timeline-item-head-gray"
        />
        <div
          class="ant-timeline-item-content"
        >
          <div
            style="font-weight: 500;"
          >
            19.9 °C
          </div>
        </div>
      </li>
      <li
        class="ant-timeline-item ant-timeline-item-right"
      >
        <div
          class="ant-timeline-item-label"
        >
          04:00
        </div>
        <div
          class="ant-timeline-item-tail"
        />
        <div
          class="ant-timeline-item-head ant-timeline-item-head-gray"
        />
        <div
          class="ant-timeline-item-content"
        >
          <div
            style="font-weight: 500;"
          >
            20 °C
          </div>
        </div>
      </li>
      <li
        class="ant-timeline-item ant-timeline-item-right"
      >
        <div
          class="ant-timeline-item-label"
        >
          05:00
        </div>
        <div
          class="ant-timeline-item-tail"
        />
        <div
          class="ant-timeline-item-head ant-timeline-item-head-gray"
        />
        <div
          class="ant-timeline-item-content"
        >
          <div
            style="font-weight: 500;"
          >
            19.9 °C
          </div>
        </div>
      </li>
      <li
        class="ant-timeline-item ant-timeline-item-right"
      >
        <div
          class="ant-timeline-item-label"
        >
          06:00
        </div>
        <div
          class="ant-timeline-item-tail"
        />
        <div
          class="ant-timeline-item-head ant-timeline-item-head-gray"
        />
        <div
          class="ant-timeline-item-content"
        >
          <div
            style="font-weight: 500;"
          >
            19.7 °C
          </div>
        </div>
      </li>
      <li
        class="ant-timeline-item ant-timeline-item-right"
      >
        <div
          class="ant-timeline-item-label"
        >
          07:00
        </div>
        <div
          class="ant-timeline-item-tail"
        />
        <div
          class="ant-timeline-item-head ant-timeline-item-head-gray"
        />
        <div
          class="ant-timeline-item-content"
        >
          <div
            style="font-weight: 500;"
          >
            19.3 °C
          </div>
        </div>
      </li>
      <li
        class="ant-timeline-item ant-timeline-item-right"
      >
        <div
          class="ant-timeline-item-label"
        >
          08:00
        </div>
        <div
          class="ant-timeline-item-tail"
        />
        <div
          class="ant-timeline-item-head ant-timeline-item-head-gray"
        />
        <div
          class="ant-timeline-item-content"
        >
          <div
            style="font-weight: 500;"
          >
            19.2 °C
          </div>
        </div>
      </li>
      <li
        class="ant-timeline-item ant-timeline-item-right"
      >
        <div
          class="ant-timeline-item-label"
        >
          09:00
        </div>
        <div
          class="ant-timeline-item-tail"
        />
        <div
          class="ant-timeline-item-head ant-timeline-item-head-gray"
        />
        <div
          class="ant-timeline-item-content"
        >
          <div
            style="font-weight: 500;"
          >
            19.2 °C
          </div>
        </div>
      </li>
      <li
        class="ant-timeline-item ant-timeline-item-right"
      >
        <div
          class="ant-timeline-item-label"
        >
          10:00
        </div>
        <div
          class="ant-timeline-item-tail"
        />
        <div
          class="ant-timeline-item-head ant-timeline-item-head-gray"
        />
        <div
          class="ant-timeline-item-content"
        >
          <div
            style="font-weight: 500;"
          >
            18.7 °C
          </div>
        </div>
      </li>
      <li
        class="ant-timeline-item ant-timeline-item-last ant-timeline-item-right"
      >
        <div
          class="ant-timeline-item-label"
        >
          11:00
        </div>
        <div
          class="ant-timeline-item-tail"
        />
        <div
          class="ant-timeline-item-head ant-timeline-item-head-gray"
        />
        <div
          class="ant-timeline-item-content"
        >
          <div
            style="font-weight: 500;"
          >
            18.5 °C
          </div>
        </div>
      </li>
    </ul>
    <ul
      class="ant-timeline ant-timeline-label css-dev-only-do-not-override-12jzuas"
      style="flex: 1;"
    >
      <li
        class="ant-timeline-item ant-timeline-item-last ant-timeline-item-left"
      >
        <div
          class="ant-timeline-item-label"
        >
          12:00
        </div>
        <div
          class="ant-timeline-item-tail"
        />
        <div
          class="ant-timeline-item-head ant-timeline-item-head-green"
        />
        <div
          class="ant-timeline-item-content"
        >
          <div
            style="font-weight: 500;"
          >
            18.3 °C
          </div>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
